### PR TITLE
Add if/in/varlist to %browse

### PR DESCRIPTION
**Pull request by kylebarron**
Continuation of #88

### CODE REVIEW

File: stata_kernel/ado/_StataKernelExportDelim.ado [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        tempvar touse sumtouse
+        gen byte `touse' = `if'
+        gen `stype' `sumtouse' = sum(`touse')
+        local last = `=`sumtouse'[_N]'
+
+        gettoken f in: in, p(/)
+        gettoken s l:  in, p(/)
+
+        if ( "`l'" == "" ) {
+            local ifin (`sumtouse' == `f')
```
Comment (mcaceresb) [inline stata_kernel/ado/_StataKernelExportDelim.ado]: This was missing ``& `touse'`` at the end.

File: stata_kernel/ado/_StataKernelExportDelim.ado [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        }
+        else {
+            local ifin (`sumtouse' >= `f') & (`sumtouse' <= `l') & `touse'
+        }
+    }
+    else if ( "`in'" != "" ) {
+        local ifin in `in'
+    }
+    else {
+        local ifin
```
Comment (mcaceresb) [inline stata_kernel/ado/_StataKernelExportDelim.ado]: I think this can default to `local ifin in 1 / 200`, right?

**Comment by mcaceresb** [comment]
Here's a version that  falls back on `list`

```
program _StataKernelExportDelim
    syntax [varlist] [if/] [in/] using, [*]

    if ( !`=_N > 0' ) error 2000

    if ( "`in'" == "" ) local in 1/200

    if ( "`if'" != "" ) {
        local stype = cond(`=_N' < maxlong(), "long", "double")
        tempvar touse sumtouse index
        gen byte `touse' = `if'
        gen `stype' `sumtouse' = sum(`touse')
        gen `stype' `index' = _n
        local last = `=`sumtouse'[_N]'
        gettoken f in: in, p(/)
        gettoken s l:  in, p(/)
        if ( "`l'" == "" ) {
            local ifin (`sumtouse' == `f') & `touse'
        }
        else {
            local ifin (`sumtouse' >= `f') & (`sumtouse' <= `l') & `touse'
        }
    }
    else {
        local ifin in `in'
    }

    qui export delimited `index' `varlist' `using' `ifin', replace `options'
    list `varlist' `ifin'
end
```

Then you can model the `_browse` magic after head to get the Stata index and send `rest` as fallback. I think it's better to let Stata handle the syntax parsing in this case, and just spit out the Stata error message in `argparse` and show the Stata syntax in the help, but that might be a matter of taste, I suppose.

**Comment by kylebarron** [comment]
Sure. The only thing I want to make sure is ok is if there aren't 200 observations in the dataset. Like with
```
sysuse auto
%browse
```
Will that automatically add `in 1/200`? Then Stata will give an index error...

**Comment by mcaceresb** [comment]
```
if ( "`in'" == "" ) local in 1/`=min(200, _N)'
```

**Comment by kylebarron** [comment]
ah brilliant!

**Comment by kylebarron** [comment]
Even when `in` is provided though, you should still do ``in 1/`=min(`in', _N)``, because it's possible `if` could subset to a very small number of people, and then `in` wouldn't be able to do, say, `1/100`.

**Comment by mcaceresb** [comment]
`if _n == 1 in 1/100` should not give an error in the program.

`in 1/100` if there are only 100 obs would give an error, but I think that, philosophically, that _should_ give an error, but it's your call.

**Pull request closed without merging**